src/common/logger.hpp

```diff
@@ -102,4 +102,32 @@ void flush();
 */
 #define SLIC_DEBUG_ROOT(rank, msg) SLIC_DEBUG_IF(rank == 0, msg)
 
+/*!
+*****************************************************************************
+* \brief Macro that logs given error message only on rank 0 if EXP is true.
+*****************************************************************************
+*/
+#define SLIC_ERROR_IF_ROOT(EXP, rank, msg) SLIC_ERROR_IF((EXP) && (rank == 0), msg)
```

Comment:
`SLIC_ERROR_ROOT_IF` may be better because this kind of sounds like it should print if the root. Any thoughts @white238 ?

Reply:
I agree.